VideoLocker/src/test/java/org/edx/mobile/test/AuthenticationTests.java

```diff
@@ -0,0 +1,135 @@
+package org.edx.mobile.test;
+
+import com.google.gson.JsonObject;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+
+import org.edx.mobile.http.OauthRefreshTokenAuthenticator;
+import org.edx.mobile.module.prefs.PrefManager;
+import org.edx.mobile.test.util.MockDataUtil;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.robolectric.annotation.Config;
+
+import java.io.IOException;
+
+import static org.edx.mobile.test.util.OkHttpTestUtil.defaultClient;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Config(sdk = 18)
+public final class AuthenticationTests extends BaseTestCase {
+
+    private static final String API_HOST_URL = "API_HOST_URL"; // Config key for API host url
+
+    @Rule
+    public final MockWebServer mockServer = new MockWebServer();
+
+    private OkHttpClient client = defaultClient();
+
+    @Before
+    public void setUp() throws Exception {
+        PrefManager pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+        pref.put(PrefManager.Key.AUTH_JSON, MockDataUtil.getMockResponse("post_oauth2_access_token"));
+        mockServer.setDispatcher(dispatcher);
+        super.setUp();
+    }
+
+    @Override
+    protected JsonObject generateConfigProperties() throws IOException {
+        // Add the mock host url in the test config properties
+        JsonObject properties = super.generateConfigProperties();
+        properties.addProperty(API_HOST_URL, mockServer.url("/").toString());
+        return properties;
+    }
+
+    @Test
+    public void testAuthenticate_forExpiredAccessToken() throws Exception {
+        // create a client with the authenticator
+        client = client.newBuilder()
+                .authenticator(new OauthRefreshTokenAuthenticator(context))
+                .build();
+
+        // Build a new dummy request to trigger authenticator
+        Request request = new Request.Builder()
+                .url(mockServer.url("/dummy/endpoint/"))
+                .header("Authorization", "expired_token")
+                .build();
+
+        // Make request
+        Response response = client.newCall(request).execute();
+        assertEquals(200, response.code());
+        assertEquals("Bearer dummy", response.request().header("Authorization"));
```

Comment:
There are additional calls being made by the authenticator even though we mocked a single requests.